subt/main.py

```diff
@@ -340,13 +322,11 @@ def wait(self, dt):  # TODO refactor to some common class
     def play_system_track(self):
         print("SubT Challenge Ver1!")
         self.go_straight(2.5)  # go to the tunnel entrance
-        dist = self.follow_wall(radius=RADIUS, right_wall=self.use_right_wall, stop_on_artf_count=1,
-                                search_since=SEARCH_TIME_BEGIN,
-                                timeout=SEARCH_TIME_END)
+        dist = self.follow_wall(radius=self.dist, right_wall=self.use_right_wall, timeout=self.timeout/2)
         print("Going HOME")
         self.turn(math.radians(90), speed=-0.1)  # it is safer to turn and see the wall + slowly backup
         self.turn(math.radians(90), speed=-0.1)
-        self.follow_wall(radius=RADIUS, right_wall=not self.use_right_wall, timeout=RETURN_TIMEOUT, dist_limit=dist+1)
+        self.follow_wall(radius=self.dist, right_wall=not self.use_right_wall, timeout=self.timeout, dist_limit=dist+1)
```

Comment:
here is example of "incompatibility" radius=self.dist vs. dist_limit=dist+1 ... "dist" is not good :(

Reply:
`timeout=self.timeout` - I am not sure if this is safe, but probably the best option

Reply:
> `timeout=self.timeout` - I am not sure if this is safe, but probably the best option

Should the command line `--timeout` be changed to `--search-time`? Or I probably don't understand fully the comment.

Reply:
this is already solved by 2.5* scaling, thanks (before I overlooked that you used half for search time). Let's use it as it is now to get some "feel" about it). For me OK.